### PR TITLE
Action tests

### PR DESCRIPTION
**Pull request by cognifloyd**
Add tests for all actions.

To do this, we reuse the hvac testing infra to setup consul/vault for our tests.

Also, we drop the broken python2.7 tests and update the CI config to use the latest from StackStorm-Exchange/ci.

~This depends on~ This demonstrates how StackStorm-Exchange/ci#101 will behave by running the `setup_testing_env.sh` script immediately following `dependencies`.

### CODE REVIEW

**Comment by cognifloyd** [comment]
I dropped 2.7 tests due to: https://github.com/StackStorm/discussions/issues/40#issuecomment-769992629

3.6 tests are failing because StackStorm-Exchange/ci#101 hasn't been merged yet.

File: tests/test_action_create_token.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            # meta=None,
+            # no_parent=False,
+            # display_name=None,
+            # num_uses=None,
+            # no_default_policy=False,
+            # ttl=None,
+            # orphan=False,
+            # wrap_ttl=None,
+        )
+        assert result["auth"]["client_token"]
```
Comment (nmaludy) [inline tests/test_action_create_token.py]: using asserts like this is going to give you some bad error messages, I think you want to use something like self.asseetEquals()

**Review by nmaludy — CHANGES_REQUESTED** [review]

File: tests/test_action_create_token.py (hunk 1 above)
Comment (cognifloyd) [reply tests/test_action_create_token.py]: Ah yes. I prefer pytest, but this is unittest. Fixed.

File: tests/vault_action_tests_base.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    secret_v1 = False
+    secret_v2 = False
+    default_token_lease = "1h"
+
+    # in setUp() and tearDown(), explicitly handle super()-like calls.
+    #  - HvacIntegrationTestCase does not call super().
+    #  - BaseActionTestCase does call super().
+
+    def setUp(self):
+        HvacIntegrationTestCase.setUp(self)
```
Comment (blag) [inline tests/vault_action_tests_base.py]: Let's use `super()` and MRO here instead of calling this explicitly.

```suggestion
        # Calls HvacIntegrationTestCase.setUp(self)
        super().setUp()
```

File: tests/vault_action_tests_base.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        # self.client = create_client(token=self.manager.root_token)
+        # also mocks hvac.utils.warnings which we might not want.
+        BaseActionTestCase.setUp(self)
```
Comment (blag) [inline tests/vault_action_tests_base.py]:
```suggestion
        # Calls BaseActionTestCase.setUp(self)
        super(HvacIntegrationTestCase, self).setUp()
```

File: tests/vault_action_tests_base.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        if self.secret_v2 and "kvv2/" not in mounted_secrets_engines:
+            self.client.enable_secret_backend(
+                backend_type="kv",
+                mount_point="kvv2",
+                options=dict(version=2),
+            )
+
+    def tearDown(self):
+        BaseActionTestCase.tearDown(self)
+        HvacIntegrationTestCase.tearDown(self)
```
Comment (blag) [inline tests/vault_action_tests_base.py]:
```suggestion
        # Calls BaseActionTestCase.tearDown(self)
        super(HvacIntegrationTestCase, self).tearDown()
        # Calls HvacIntegrationTestCase.tearDown(self)
        super().tearDown()
```

**Review by blag — CHANGES_REQUESTED** [review]
I would suggest using `super()` where we can. Here's a [write up](https://rhettinger.wordpress.com/2011/05/26/super-considered-super/) from Raymond Hettinger (famous Pythonista) about `super()`.

TL;DR: It leads to less code churn as parent classes change.

Other than that, this looks great!

File: tests/vault_action_tests_base.py (hunk 3 above)
Comment (cognifloyd) [reply tests/vault_action_tests_base.py]: Huh. Calling super on the class that doesn't call it is a fascinating way to fix the issue. Cool.

**Review by cognifloyd — COMMENTED** [review]
Fascinating. I wasn't sure how to deal with calling super() when one of the parents didn't also call super(). Still using `super(SuperCallLessClass, self)` is a nicer work around. I've pushed the changes and adjusted the comments to help future me understand what's going on.

File: tests/vault_action_tests_base.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            self.client.enable_secret_backend(
+                backend_type="kv",
+                mount_point="kvv2",
+                options=dict(version=2),
+            )
+
+    def tearDown(self):
+        super(VaultActionTestCase, self).setUp()
+        # HvacIntegrationTestCase does not call super(), so we take care of that.
+        super(HvacIntegrationTestCase, self).setUp()
```
Comment (blag) [inline tests/vault_action_tests_base.py]: NAK - this calls `setUp()` again, not `tearDown()` like it should.

```suggestion
        super(VaultActionTestCase, self).tearDown()
        # HvacIntegrationTestCase does not call super(), so we take care of that.
        super(HvacIntegrationTestCase, self).tearDown()
```

**Review by blag — CHANGES_REQUESTED** [review]
One small mistake needs to be fixed.

**Review by blag — APPROVED** [review]
Awesome, thanks for taking this on!
